src/query_plan/operation.rs

```diff
@@ -537,6 +542,42 @@ impl NormalizedSelection {
     pub(crate) fn has_defer(&self) -> Result<bool, FederationError> {
         todo!()
     }
+
+    fn collect_used_fragment_names(&self, aggregator: &mut Arc<HashMap<Name, i32>>) {
```

Comment:
Yeah. I originally wrapped it in Arc as this calculated info is actually stored on fragment itself, since we are not doing that anymore then there is no point in wrapping it.